### DISCUSSION TITLE
Where is the data from the `--details` flag stored?

### DISCUSSION THREAD

**Discussion by jmgilman** [vmware-tanzu/velero#5912] [Community support Q&A] [answered]
```
velero describe backup <name> --details
```

I'm using the above command to inspect a backup. In particular, this backup contains native volume snapshots (from EBS). It returns a list of PVCs along with their respective snapshot IDs. 

```
kubectl get backup <name> -o yaml
```

Running the above command, I would expect to see the same data available from the velero CLI. Yet, I don't see any information about the snapshots. I noticed there are other mysterious fields the velero CLI returns like a list of resources, hooks, etc. Where is the CLI getting all of this additional information?

**Comment by sseago** [comment, marked as answer]
Kubectl only shows information that's stored in the velero backup CR. The velero client `describe` command pulls in additional information stored in the object store (backup storage location) associated with the backup. The resource list, for example, is not part of the kubernetes resource, but it is in object store. Same with PVC list and snapshot ides. All in all, the backup CR tends to contain summary information, but the describe CLI is able to pull more detailed information for the backup that is contained in the object store.

**Reply by jmgilman** [reply]
Thanks, that's a helpful clarification. Does the object store happen to use an open-source format? I need access to the snapshot IDs for EBS because I'm realizing that a "Complete" status for a backup doesn't mean the snapshots have actually completed.

**Reply by sseago** [reply]
If you're wondering about inspecting backup contents directly vs. using the velero client, that's generally all stored as gzipped json (and the backup itself is a gzipped tar file of kubernete resources), so if you can parse json, tar, and gz, you should be able to read those files in a tool outside of velero.
